libcaf_net/test/endpoint_manager.cpp

```diff
@@ -81,14 +83,16 @@ class dummy_transport {
 
   template <class Manager>
   error init(Manager& manager) {
-    write_buf_.insert(write_buf_.end(), hello_test.begin(), hello_test.end());
+    write_buf_.insert(write_buf_.end(),
+                      reinterpret_cast<const byte*>(hello_test.begin()),
```

Comment:
This code is brittle, because it relies on `iterator` to be the same as `pointer` in order to be able to `reinterpret_cast` iterators safely. We should make as few assumptions as possible when writing code and shouldn't rely on implementations details.

Also, we can get around the casts entirely:

```c++
auto test_bytes = as_bytes(make_span(hello_test));
write_buf_.insert(write_buf_.end(), test_bytes.begin(), test_bytes.end());
```